src/borg/archiver/extract_cmd.py

```diff
@@ -71,6 +71,7 @@ def do_extract(self, args, repository, manifest, archive):
                 logging.getLogger("borg.output.list").info(remove_surrogates(item.path))
             try:
                 if dry_run:
+                    print(f"- {item.path}")
```

Comment:
check the meaning of the flags (see our docs) and try running borg create with some excludes. borg extract should flag similarly to create.

Reply:
also be careful with print, print outputs to stdout and borg extract also has the option to write extracted file contents to stdout - these 2 shall not be mixed up.

Reply:
Apologies for the oversight. Trying a new fix and uploading soon for review.